opal/mca/btl/tcp/btl_tcp_endpoint.c

```diff
@@ -406,12 +397,30 @@ mca_btl_tcp_endpoint_send_blocking(mca_btl_base_endpoint_t* btl_endpoint,
 
 static int mca_btl_tcp_endpoint_send_connect_ack(mca_btl_base_endpoint_t* btl_endpoint)
 {
-    /* send process identifier to remote endpoint */
+
     opal_process_name_t guid = opal_proc_local_get()->proc_name;
 
```

Comment:
asprintf the entire id and send it at once ? Or extend the blocking API to take an iovec (this second approach seems cleaner).